### PR TITLE
remove unused import statements

### PR DESCRIPTION
**Pull request by csimons**
In addition to cluttering the namespace, unused imports generate compiler warnings, introducing noise and obscuring more important compiler warnings.  This change removes them.

### CODE REVIEW

**Comment by KostyaSha** [comment]
Such changes without any future automated verification doesn't make any sense because any new contributor wouldn't remove unused imports. In github-plugin automated checks successfully picked @jglick forgotten unused imports.
So technically :-1: for making code cleanups without future protection for such things. I proposed conservative style checks in dev list, but as usual any dev list discussion has no any results.

**Comment by KostyaSha** [comment]
Cross-linking again coding style clean-ups to dev list https://groups.google.com/forum/#!topic/jenkinsci-dev/8fjvXGYbFJ4

**Comment by jtnord** [comment]
@KostyaSha even if we can't protect against regressions I don't see why we should fail to accept this.
@csimons put in time to do this and it helps developers now. 

That is accepting this contribution is not adding to technical debt -  and by accepting it no one is making any guarantees that we won't accept PRs or commits with unused imports in the future.  If we don't start improving the codebase bit by bit (and it may be 2 steps forward and one step back) then we will just degrade into a sea of uncontrollable style mess as no one will ever be able to improve something until we have tooling in place.  Getting that tooling agreed yet and in place is not going to be a quick solution - yet alone getting consensus on what is an appropriate thing to enforce.

so basically I am :+1: on this change if it passes the build.

**Comment by batmat** [comment]
:+1: to accept. 
@KostyaSha it would be wasteful to reject *just because* future useless imports are not catched.

**Comment by KostyaSha** [comment]
@jtnord i started discussion in maillist _before_ to catch such problems in automated way to exclude such barriers for new-comers. It sounds like i can create automated tool that will PR such commits after they merged and i can't make PR validation before the merge because email discussion will have no results and there is no policy that unused imports should be removed. Note, under `*` imports there are many unused imports. Doesn't this sound weird? :)

@batmat i of course :+1: for any code clean-ups, but people shouldn't waste time on corner cases situations.

**Comment by olivergondza** [comment]
@csimons, please squash this into a single commit and I will merge it.

**Comment by KostyaSha** [comment]
@olivergondza why? It will create merge problems for other PRs as they will be forced to merge the whole commit.

**Comment by olivergondza** [comment]
Well, I do not really see a point in creating 250+ commits to remove 250+ lines. Even the commit message just reiterates the exact change.

@KostyaSha I give up. We spent lot more time with this squash or not to squash discussion than any outcome can compensate in reasonable future. Also, this is going to cause the same number of merge conflicts despite the number of commits it is chopped into. Actually, after squashed you can resolve all of them at once and not commit-by-commit.

**Comment by KostyaSha** [comment]
@olivergondza i give up also :) I thought idea was to have logical commits. I.e. when you are looking at commit history for class it will be easier to follow single commit change that removed unused import than diff with many-many files in party. @olivergondza i think that single file change in this case will require you to pick this changes before applying new changes for some "feature change" (that logically other commit).

**Comment by csimons** [comment]
@olivergondza This has been squashed.

**Comment by csimons** [comment]
@KostyaSha Yeah, the use of squashing is an interesting question.  In this case, the commits are self-contained in a technical sense, with none breaking the build, but at the same time they're all logically the same idea and made for the same (single) purpose.  Additionally, it seems undesirable to make people scroll through 10+ pages of these commits in order to view the history that came before.

**Comment by KostyaSha** [comment]
As i see IDE marks any import that has mention javadoc as used. See, http://checkstyle.sourceforge.net/config_imports.html#UnusedImports 
`Optionally: it is referenced in Javadoc comments. This check is on by default, but it is considered bad 
practice to introduce a compile time dependency for documentation purposes only. As an example, the
 import java.util.Date would be considered referenced with the Javadoc comment {@link Date}. The alternative to avoid introducing a compile time dependency would be to write the Javadoc comment as {@link java.util.Date}.`

**Comment by olivergondza** [comment]
@KostyaSha, have we removed imports that are needed by javadoc?

**Comment by KostyaSha** [comment]
@olivergondza I have not tested. Experimenting with checkstyle rules on plugin, just kept here for info that may by 2 types of unused imports. @csimons was unused imports message generated by `javac`?

**Comment by csimons** [comment]
@KostyaSha @olivergondza To my knowledge, no classes needed to resolve Javadoc links were removed by this change.  In the event we come across such a case, I agree with the excerpt above that class names should be fully-qualified in the Javadoc so as not to introduce an unnecessary compilation dependencies.

@KostyaSha Nope, these were found using a static analysis tool.
